src/app/resources/iis_scripts/admin_api_util.ps1

```diff
@@ -0,0 +1,120 @@
+param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('install', 'permission-check')]
+    [string]
+    $command,
+
+    [string]
+    $downloadFrom,
+
+    ## Not doing anything with this parameter yet
+    [string]
+    $sessionId,
+
+    [string]
+    $serviceName = "Microsoft IIS Administration",
+
+    [string]
+    $iisAdminGroup = "WAC-IISAdmin"
+)
+
+$ErrorActionPreference = "Stop"
+
+######################## Param checks ###############################
+$install = $command -eq 'install'
+
+if ($install -and $(!$downloadFrom)) {
+    throw "Missing download location for install command"
+}
+
+######################## Utilities ###############################
+function InstallAPI {
+    $installer = Join-Path $env:TEMP iis-administration-setup.exe
+    Invoke-WebRequest $downloadFrom -OutFile $installer
+    & $installer /s
+
+    $waitPeriod = 1
+    $remainingTime = 600
+
+    while (!(Get-Service $serviceName -ErrorAction SilentlyContinue | Where-Object {$_.Status -eq "Running"})) {
+        Start-Sleep -Seconds $waitPeriod
+        $remainingTime -= $waitPeriod
+        if ($remainingTime -lt 0) {
+            throw "timeout waiting for service to start"
+        }
+    }
+}
+
+function EnsureAdminGroup($user) {
+    New-LocalGroup -Name $iisAdminGroup -ErrorAction SilentlyContinue | Out-Null
+    Add-LocalGroupMember -Group $iisAdminGroup -Member $user -ErrorAction SilentlyContinue | Out-Null
+}
+
+####################### Main script ################################
+
+## TODO: Consider creating a lock around this script because it is not concurrency safe
+
+if ($install) {
+    InstallAPI
+}
+
+$user = $(whoami)
+EnsureAdminGroup $user
+
+$service = Get-WmiObject win32_service | Where-Object {$_.name -eq $serviceName}
+if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
+    $workingDirectory = $service.StartInfo.WorkingDirectory
+} else {
+    $proc = Get-Process -id $service.ProcessId
+    $workingDirectory = Split-Path ($proc.Modules | Where-Object { $_.ModuleName -eq "Microsoft.IIS.Administration.dll" }).FileName
+}
+
+$configLocation = [System.IO.Path]::Combine($workingDirectory, "config", "appsettings.json")
+$config = Get-Content -Raw -Path $configLocation | ConvertFrom-Json
+
+if ($install) {
+    $config.security.users.administrators = @( $iisAdminGroup )
+    $config.security.users.owners = @( $iisAdminGroup )
+    $updateConfig = $true
+} else {
+    if (!$config.security.users.administrators.Contains($user)) {

```

Comment:
I don't understand why this logic would be different whether installing or not. It seems like it should just be checks to see if the desired principal exists in _owners_ or _administrators_ and then add it.

Reply:
I am assuming if the Admin API was already installed. I would like the original owners/administrators still have access to have API without using our groups

Reply:
I don't think that ensuring that the group that we create is part of the owners and administrators roles would remove any existing access of the installation. The WAC scenario would be more consistent if the IIS Administrator Owners group was added to the iis administration API roles regardless of whether the API was just installed or not.

Reply:
Sure, I was kind of wondering between the two approaches. This may be cleaner. We should just communicate to clients that if WAC is involved we always manage permission with local groups